editor/selectors.js

```diff
@@ -12,6 +12,9 @@ import { serialize, getBlockType } from '@wordpress/blocks';
 import { __ } from '@wordpress/i18n';
 import { addQueryArgs } from '@wordpress/url';
 
+const REGEXP_EMPTY_CONTENT = /^<!--(.)*>(\s)*<p[^>]*>(&nbsp|\s|<br[^>]*>)*<\/p>(\s)(.)*-->$/mg;
```

Comment:
`.*` are prone to greediness, which results in issues like this:

https://regex101.com/r/64lOJ1/1

http://www.regular-expressions.info/repeat.html